lib/scala/filewatcher/src/main/java/org/enso/filewatcher/DefaultWatcher.java

```diff
@@ -149,27 +168,27 @@ private void exceptionCallback(Throwable cause) {
     exceptionCallback.accept(watcherError);
   }
 
-  private void registerWatchService(Path path) {
-    LOGGER.debug("Registering watch service for subdir {}", path);
+  private void registerWatchService(Path dir) {
+    LOGGER.debug("Registering watch service for subdir {}", dir);
     try {
       var watchKey =
-          path.register(
+          dir.register(
               watchService,
               StandardWatchEventKinds.ENTRY_CREATE,
               StandardWatchEventKinds.ENTRY_MODIFY,
               StandardWatchEventKinds.ENTRY_DELETE,
               StandardWatchEventKinds.OVERFLOW);
-      watchedDirs.put(path, watchKey);
+      watchedDirs.put(dir, watchKey);
```

Comment:
```suggestion
      watchedDirs.put(dir, watchKey);
      assert watchKey.watchable() == dir;
```
if this `assert` holds (and I believe it does), then you can simplify the [matchingEntry = watchedDirs.entrySet().stream()... logic]() to just:
```java
dir = (Path)key.watchable();
```

Reply:
Right. Stream filtering removed in 2eb00920639dbd650be9ac6972d98dbf34acb798